src/middle/ssa/ssastorage.rs

```diff
@@ -249,14 +249,9 @@ impl SSAStorage {
         let mut adjacent = Vec::new();
         let mut walk = self.g.neighbors_directed(node, direction).detach();
         while let Some((edge, othernode)) = walk.next(&self.g) {
-            if data {
-                if let EdgeData::Data(i) = self.g[edge] {
-                    adjacent.push((i, othernode))
-                }
-            } else {
-                if let EdgeData::Control(i) = self.g[edge] {
-                    adjacent.push((i, othernode));
-                }
+            match self.g[edge] {
```

Comment:
Doesn't it:
`data == true` implies edge type is EdgeData::Data
`data == false` implies edge type is EdgeData::Control?
If no then it's a mistake, right